src/constraint_system/ecc/scalar_mul/fixed_base.rs

```diff
@@ -313,6 +313,7 @@ mod tests {
         assert!(res.is_ok());
     }
 
+    #[allow(non_snake_case)] // NOTE: We want to use capital `G` and `H`.
```

Comment:
We should move this lint to `lib.rs` and allow it for the entire lib. As sometimes we want to use capital letter variables.

Reply:
Honestly, I'm of the opinion you should not use non-Rust conventions for naming and it should only be used in cases where you are trying to match some specification in which case it's more important to get the spec and impl as close to each other as possible. Like code complexity lints, it should be disabled in a more precise way than just everywhere. I just did the smallest change here to remove the issue.

Reply:
So while I understand your POV, and i respect it, with EC libs, sometimes it's much much easier to read code when you can identify variables and differenciate scalars and points just by seeing whether they're capital or non-capital letters. 

Let's keep the lint for now. But I'm slightly biased on this one to move the lint to be general for the entire library..